wallet/src/main/java/com/android/identity_credential/wallet/MainActivity.kt

```diff
@@ -0,0 +1,477 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.android.identity.util.Logger
+import com.android.identity_credential.wallet.ui.theme.IdentityCredentialTheme
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+class MainActivity : ComponentActivity() {
+
+    companion object {
+        private const val TAG = "MainActivity"
+    }
+
+    private lateinit var application: WalletApplication
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        application = getApplication() as WalletApplication
+
+        setContent {
+            IdentityCredentialTheme {
+                // A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colorScheme.background
+                ) {
+                    val navController = rememberNavController()
+                    NavHost(navController = navController, startDestination = "MainScreen") {
+                        composable("MainScreen") {
+                            MainScreen(navController)
+                        }
+                        composable("AboutScreen") {
+                            AboutScreen(navController)
+                        }
+                        composable("AddToWalletScreen") {
+                            AddToWalletScreen(navController)
+                        }
+                        composable("CredentialInfo/{credentialId}") { backStackEntry ->
+                            CredentialInfoScreen(navController,
+                                backStackEntry.arguments?.getString("credentialId")!!)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    @Composable
+    fun MainScreen(navigation: NavHostController) {
+        val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+        val scope = rememberCoroutineScope()
+        ModalNavigationDrawer(
+            drawerState = drawerState,
+            drawerContent = {
+                ModalDrawerSheet {
+                    Text("Wallet", modifier = Modifier.padding(16.dp))
+                    Divider()
+                    NavigationDrawerItem(
+                        icon = { Icon(imageVector = Icons.Filled.Add, contentDescription = null) },
+                        label = { Text(text = "Add to Wallet") },
```

Comment:
imo would be more clear if it said something along the lines of "Add new Credential to Wallet"

Reply:
This is displayed in the UI in a pretty narrow drawer so I'd like to keep it short.